packages/@aws-cdk/aws-elasticloadbalancingv2-actions/lib/cognito-action.ts

```diff
@@ -0,0 +1,89 @@
+import * as cognito from '@aws-cdk/aws-cognito';
+import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
+import { Duration } from '@aws-cdk/core';
+
+/**
+ * Properties for AuthenticateCognitoAction
+ */
+export interface AuthenticateCognitoActionProps {
+  /**
+   * What action to execute next
+   */
+  readonly next: elbv2.ListenerAction;
+
+  /**
+   * The Amazon Cognito user pool.
+   */
+  readonly userPool: cognito.IUserPool;
```

Comment:
Both the userpool domain and userpool clients can get you the underlying user pool. Might need to add a getter in the cognito module.

It's probably not correct to pass domain and clients of a different user pool from this, so omitting this might reduce human error.

Reply:
I am wondering whether a better solution is to make the client optional and automatically generate it if not supplied, as its configuration is quite finicky.

But for now, I just want to get this out there and start collecting feedback and bug reports.

Reply:
Also, `IUserPool` definitely does not have a method to get all clients (could it even?), nor does an `IUserPoolClient` have a way to get the UserPool associated with it.

Reply:
> does an IUserPoolClient have a way to get the UserPool associated with it.

This is the first use case, so we'll need to add it.

> I am wondering whether a better solution is to make the client optional and automatically generate it if not supplied, as its configuration is quite finicky.

I like it. Let's do this.